cirq-core/cirq/ops/controlled_gate.py

```diff
@@ -241,26 +247,26 @@ def get_prefix(control_vals):
                 control_vals_str = ''.join(map(str, sorted(control_vals)))
                 return f'C{control_vals_str}'
 
-        return ''.join(map(get_prefix, self.control_values)) + str(self.sub_gate)
+        return ''.join(map(get_prefix, self.control_values.arrangements())) + str(self.sub_gate)
 
     def __repr__(self) -> str:
-        if self.num_controls() == 1 and self.control_values == ((1,),):
+        if self.num_controls() == 1 and self.control_values.are_same_value(1):
             return f'cirq.ControlledGate(sub_gate={self.sub_gate!r})'
 
-        if all(vals == (1,) for vals in self.control_values) and set(self.control_qid_shape) == {2}:
+        if self.control_values.are_same_value(1) and set(self.control_qid_shape) == {2}:
             return (
                 f'cirq.ControlledGate(sub_gate={self.sub_gate!r}, '
                 f'num_controls={self.num_controls()!r})'
             )
         return (
             f'cirq.ControlledGate(sub_gate={self.sub_gate!r}, '
-            f'control_values={self.control_values!r},'
+            f'control_values={self.control_values.arrangements()!r},'
```

Comment:
Does this still round-trip? Given there's no type information about which subclass of `ControlValues` it uses, I'm not seeing how it could.